mobile/components/Avatar/index.js

```diff
@@ -1,18 +1,32 @@
 // @flow
 import React, { Component } from 'react';
 import { AvatarImage } from './style';
+import compose from 'recompose/compose';
+import { withNavigation } from 'react-navigation';
+import { TouchableHighlight } from 'react-native';
 
 type AvatarProps = {
   src: string,
   size: number,
   radius: number,
+  navigate?: Function,
 };
 
-export default class Avatar extends Component<AvatarProps> {
+class Avatar extends Component<AvatarProps> {
   render() {
-    const { src, size, radius } = this.props;
+    const { src, size, radius, navigate } = this.props;
     let source = src ? { uri: src } : {};
 
+    if (navigate) {
+      return (
+        <TouchableHighlight onPress={() => navigate()}>
```

Comment:
I'm not sure it makes sense to bake this into the `Avatar` component, I'm sure there'll be cases where we don't want it to be clickable.

Instead, let's move this outside to the individual uses so it's clear where it's clickable and where it isn't, and keep this component focused.

Reply:
After looking through how you use the `Avatar` component, I now see that `navigate` is actually passed from the props and not injected from the `withNavigation` call further down below, so that's actually not so bad.

Let's remove the unnecessary `withNavigation` wrapper and rename the prop `onPress`. (why should you only have to navigate on press?! not very intuitive)